tt_metal/hw/inc/circular_buffer.h

```diff
@@ -75,24 +93,51 @@ inline void setup_cb_read_write_interfaces(uint32_t tt_l1_ptr *cb_l1_base, uint3
         uint32_t fifo_page_size = circular_buffer_config_addr[3];
         uint32_t fifo_limit = fifo_addr + fifo_size;
 
-        cb_interface[cb_id].fifo_limit = fifo_limit;  // to check if we need to wrap
+        LocalCBInterface &cb_interface = get_local_cb_interface(cb_id);
+        cb_interface.fifo_limit = fifo_limit;  // to check if we need to wrap
         if (write) {
-            cb_interface[cb_id].fifo_wr_ptr = fifo_addr;
+            cb_interface.fifo_wr_ptr = fifo_addr;
         }
         if (read) {
-            cb_interface[cb_id].fifo_rd_ptr = fifo_addr;
+            cb_interface.fifo_rd_ptr = fifo_addr;
         }
-        cb_interface[cb_id].fifo_size = fifo_size;
-        cb_interface[cb_id].tiles_acked_received_init = 0;
+        cb_interface.fifo_size = fifo_size;
+        cb_interface.tiles_acked_received_init = 0;
         if (write) {
-            cb_interface[cb_id].fifo_num_pages = fifo_num_pages;
+            cb_interface.fifo_num_pages = fifo_num_pages;
         }
-        cb_interface[cb_id].fifo_page_size = fifo_page_size;
+        cb_interface.fifo_page_size = fifo_page_size;
 
         if (init_wr_tile_ptr) {
-            cb_interface[cb_id].fifo_wr_tile_ptr = 0;
+            cb_interface.fifo_wr_tile_ptr = 0;
         }
 
-        circular_buffer_config_addr += WORDS_PER_CIRCULAR_BUFFER_CONFIG;
+        circular_buffer_config_addr += UINT32_WORDS_PER_LOCAL_CIRCULAR_BUFFER_CONFIG;
+    }
+}
+
+inline void setup_remote_cb_interfaces(uint32_t tt_l1_ptr *cb_l1_base, uint32_t start_cb_index) {
+    volatile tt_l1_ptr uint32_t* circular_buffer_config_addr = cb_l1_base;
+
+    for (uint32_t cb_id = NUM_CIRCULAR_BUFFERS - 1, end_id = start_cb_index - 1; cb_id != end_id; cb_id--) {
+        volatile tt_l1_ptr uint32_t* l1_remote_cb_config_addr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(circular_buffer_config_addr[0]);
+        RemoteCBInterface& cb_interface = get_remote_cb_interface(cb_id);
+        cb_interface.fifo_start_addr = l1_remote_cb_config_addr[0];
```

Comment:
since l1_remote_cb_config_addr is a volatile, this sequence will look like:
load store
load store
load store
and therefore pay the L1 latency cost on each load.  Instead, do the loads up front into temporaries and then the stores to hide the L1 cost.  (and confirm before/after in the assembly)